driver/configuration/provider_viper.go

```diff
@@ -86,35 +86,9 @@ const (
 	ViperKeyAuthenticatorUnauthorizedIsEnabled = "authenticators.unauthorized.enabled"
 )
 
-func BindEnvs() {
-	if err := viper.BindEnv(
-		ViperKeyProxyReadTimeout,
-		ViperKeyProxyWriteTimeout,
-		ViperKeyProxyIdleTimeout,
-		ViperKeyProxyServeAddressHost,
-		ViperKeyProxyServeAddressPort,
-		ViperKeyAPIServeAddressHost,
-		ViperKeyAPIServeAddressPort,
-		ViperKeyAccessRuleRepositories,
-		ViperKeyAuthorizerAllowIsEnabled,
-		ViperKeyAuthorizerDenyIsEnabled,
-		ViperKeyAuthorizerKetoEngineACPORYIsEnabled,
-		ViperKeyMutatorCookieIsEnabled,
-		ViperKeyMutatorHeaderIsEnabled,
-		ViperKeyMutatorNoopIsEnabled,
-		ViperKeyMutatorHydratorIsEnabled,
-		ViperKeyMutatorIDTokenIsEnabled,
-		ViperKeyMutatorIDTokenJWKSURL,
-		ViperKeyAuthenticatorAnonymousIsEnabled,
-		ViperKeyAuthenticatorNoopIsEnabled,
-		ViperKeyAuthenticatorCookieSessionIsEnabled,
-		ViperKeyAuthenticatorJWTIsEnabled,
-		ViperKeyAuthenticatorOAuth2ClientCredentialsIsEnabled,
-		ViperKeyAuthenticatorOAuth2TokenIntrospectionIsEnabled,
-		ViperKeyAuthenticatorUnauthorizedIsEnabled,
-	); err != nil {
-		panic(err.Error())
-	}
+func init() {
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	viper.AutomaticEnv()
```

Comment:
This is executed properly by `viperx` on boot. No need to set/use this here.

Reply:
I'm not sure why it is called in `InitializeConfig` and not either in the init function or the separate function. This way tests that simply call `NewViperProvider` are never getting this effect. I think this should be encapsulated in the provider. Maybe even still in both places.

Reply:
Because we're initializing viper the same way in all projects. Encapsulating it in a library helps with keeping things coherent between projects. This will not change.

These configs are not useful for testing, we do not set env vars in such a way in our tests. For isolated tests, like the one you did, we can manually either invoke those two functions or export the functionality in `viperx`.